### PR TITLE
Add examples on querying records with given reference

### PR DESCRIPTION
**Pull request by david90**
refs. #97

### CODE REVIEW

**Comment by chpapa** [comment]
@david90 thanks for the work!

However, I think the guide of relation query should have two sections instead?

Section 1: Show how to query with an `account` object (instead of ID)
Section 2: If the developer got only an ID, query like this PR already do.

Can you try to revise it? Thanks.

**Comment by david90** [comment]
@chpapa updated, appended in this PR.

File: cloud-db/queries/android.md [hunk 1]
```diff
@@ -102,6 +102,56 @@ Besides the operations shown above, the following list out all operations suppor
 
 ## Relational Queries
 
+This example shows how to query all notes (`Note` record) who has an `account` field reference to a user record. In this example, we will query all notes where `account` equals to the current user.
+
+```java
+Record currentUser = Container.defaultContainer(this).getAuth().getCurrentUser(); // Get the current user
+Query noteQuery = new Query("Note").transientInclude("account")
```
Comment (chpapa) [inline cloud-db/queries/android.md]: First example should ignore transientInclude? Or leave it as a comment to say sth like // .transientInclude("account") // let's leave eager loading alone now, will explain in next section.

File: cloud-db/queries/android.md [hunk 2]
```diff
@@ -102,6 +102,56 @@ Besides the operations shown above, the following list out all operations suppor
...
+        }
+    }
+
+    @Override
+    public void onQueryError(Error error) {
+        Log.i("Record Query", String.format("Fail with reason:%s", error.getLocalizedMessage()));
+    }
+```
+
+If you haven't have the corresponding record in hand (in this example, we will use the User record `182654c9-d205-43aa-8e74-d465c830087a`), you can reference with a specify `id` in this way:
```
Comment (chpapa) [inline cloud-db/queries/android.md]: you can reference with a specific `id` without making another query in this way:

File: cloud-db/queries/ios.md [hunk 3]
```diff
@@ -110,10 +110,105 @@ NSPredicate *p =
...
+    
+    print ("Received \(results?.count) notes.")
+    
+    for note in results as! [SKYRecord] {
+        print ("Got a Note  \(note["content"])")
+    }
+}
+```
+
+If you haven't have the corresponding record in hand (in this example, we will use the User record `182654c9-d205-43aa-8e74-d465c830087a`), you can reference with a specify `id` in this way:
```
Comment (chpapa) [inline cloud-db/queries/ios.md]: add "without making another query" here too

File: cloud-db/queries/ios.md [hunk 4]
```diff
@@ -110,10 +110,105 @@ NSPredicate *p =
...
+
+```objective-c
+SKYReference *nameRef = [SKYReference referenceWithRecordID:[SKYRecordID recordIDWithCanonicalString:@"account/182654c9-d205-43aa-8e74-d465c830087a"]];
+
+NSPredicate *accountPredicate = [NSPredicate predicateWithFormat:@"account = %@", nameRef];
+
+SKYQuery *query = [SKYQuery queryWithRecordType:@"Note" predicate: accountPredicate];
+
+
+SKYDatabase *privateDB = [[SKYContainer defaultContainer] privateCloudDatabase];
```
Comment (chpapa) [inline cloud-db/queries/ios.md]: Why this example suddenly change to privateDB?

File: cloud-db/queries/js.md [hunk 5]
```diff
@@ -184,6 +184,35 @@ to get the count without fetching any records, simply set `query.limit = 0`.
...
+let query = new skygear.Query(Note);
+query.equalTo('account', new skygear.Reference(skygear.auth.currentUser));
+
+skygear.publicDB.query(query).then((r) => {
+    console.log(r);
+});
+
+```
+
+If you haven't have the corresponding record in hand (in this example, we will use the User record `182654c9-d205-43aa-8e74-d465c830087a`), you can reference with a specify `id` in this way:
```
Comment (chpapa) [inline cloud-db/queries/js.md]: add "without making another query" here too

**Review by chpapa — CHANGES_REQUESTED** [review]
Some final comments on style and details @david90  Thanks!

**Comment by david90** [comment]
Updated accordingly @chpapa

File: cloud-db/queries/android.md [hunk 6]
```diff
@@ -102,6 +102,55 @@ Besides the operations shown above, the following list out all operations suppor
 
 ## Relational Queries
 
+This example shows how to query all notes (`Note` record) who has an `account` field reference to a user record. In this example, we will query all notes where `account` equals to the current user.
+
+```java
+Record currentUser = Container.defaultContainer(this).getAuth().getCurrentUser(); // Get the current user
+Query noteQuery = new Query("Note").equalTo("account", currentUser.getId());
```
Comment (chpapa) [inline cloud-db/queries/android.md]: Can we do `.equalTo("account", currentUser)` actually?

Comment (chpapa) [inline cloud-db/queries/android.md:109]: Can we do .equalTo("account", currentUser) actually?

File: cloud-db/queries/android.md [hunk 7]
```diff
@@ -102,6 +102,55 @@ Besides the operations shown above, the following list out all operations suppor
...
+    @Override
+    public void onQueryError(Error error) {
+        Log.i("Record Query", String.format("Fail with reason:%s", error.getLocalizedMessage()));
+    }
+```
+
+If you haven't have the corresponding record in hand (in this example, we will use the User record `182654c9-d205-43aa-8e74-d465c830087a`), you can reference with a specify `id` without making another query in this way:
+
+```java
+Query noteQuery = new Query("Note").transientInclude("account")
```
Comment (chpapa) [inline cloud-db/queries/android.md]: Inconsistent to still have transientInclude in this example.

File: cloud-db/queries/ios.md [hunk 8]
```diff
@@ -110,10 +110,105 @@ NSPredicate *p =
 let p = SKYRelationPredicate(relation: SKYRelation.following(), keyPath: "_owner")
 ```
 
-### Full-text search (**NOT IMPLEMENTED**)
-
 ### References
```
Comment (chpapa) [inline cloud-db/queries/ios.md]: `Relational Queries` instead.

**Review by chpapa — CHANGES_REQUESTED** [review]
@david90 sorry 1 question and two very minor changes just spotted :(

File: cloud-db/queries/android.md (hunk 6 above)
Comment (david90) [reply cloud-db/queries/android.md:109]: @chpap No, it doesn't work that way so I have to specify `getId()` here.

**Comment by david90** [comment]
@chpapa Fixed the remaining two issues except "Android relation query equal" one.

**Review by chpapa — APPROVED** [review]

**Pull request merged**